src/arrayMethodJoin.js

```diff
@@ -1,11 +1,29 @@
 'use strict';
 
-/**
- * Implement method join
- */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    const len = this.length - 1;
+
+    for (let i = 0; i < len; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        this[i] = '';
+      }
+
+      if (separator === null) {
+        result = result + this[i] + 'null';
+      } else if (separator === undefined) {
+        result = result + this[i] + ',';
+      } else {
+        result = result + this[i] + separator;
+      }
+    }
+
+    if (this.length === 0) {
```

Comment:
не потрібно перевіряти на 0

```suggestion
    if (!this.length) {
```